### PR TITLE
chore: add help menu to the Makefile

### PR DESCRIPTION
**Pull request by andrewrynhard**
This adds a help  menu to the Makefile. It documents all build
dependencies, and how to get started.

### CODE REVIEW

File: docs/osctl/osctl_cluster_create.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ osctl cluster create [flags]
       --cidr string                 CIDR of the docker bridge network (default "10.5.0.0/24")
       --cpus string                 the share of CPUs as fraction (each container) (default "1.5")
   -h, --help                        help for create
-      --image string                the image to use (default "docker.io/autonomy/talos:v0.3.0-beta.0-16-g1b0b6b3d-dirty")
+      --image string                the image to use (default "docker.io/autonomy/installer:latest")
```
Comment (smira) [inline docs/osctl/osctl_cluster_create.md]: this looks wrong?

File: cmd/osctl/cmd/cluster.go [hunk 2]
```diff
@@ -383,7 +382,7 @@ func parseCPUShare() (int64, error) {
 }
 
 func init() {
-	clusterUpCmd.Flags().StringVar(&nodeImage, "image", "docker.io/autonomy/talos:"+version.Tag, "the image to use")
+	clusterUpCmd.Flags().StringVar(&nodeImage, "image", defaultImage(), "the image to use")
```
Comment (smira) [inline cmd/osctl/cmd/cluster.go]: I guess this should be different from `--install-image`?

File: docs/osctl/osctl_cluster_create.md (hunk 1 above)
Comment (andrewrynhard) [reply docs/osctl/osctl_cluster_create.md]: Oh whoops.

File: cmd/osctl/cmd/cluster.go (hunk 2 above)
Comment (andrewrynhard) [reply cmd/osctl/cmd/cluster.go]: Thanks for catching.

File: Makefile [hunk 3]
```diff
@@ -41,250 +25,224 @@ COMMON_ARGS += --build-arg=TOOLS=$(TOOLS)
...
 COMMON_ARGS += --build-arg=GO_VERSION=$(GO_VERSION)
-COMMON_ARGS += .
+COMMON_ARGS += --build-arg=IMAGES=$(ARTIFACTS)/images
 
-TESTPKGS ?= ./...
+all: help
+
+# Help Menu
+
+define HELP_MENU_HEADER
```
Comment (andrewrynhard) [inline Makefile]: Take a close look here. This is the Makefile help banner that should have enough to get someone started developing Talos.

File: hack/installer/entrypoint.sh [hunk 4]
```diff
@@ -137,6 +137,19 @@ case "$1" in
...
+        n )
+          TALOS_RAW="/out/${OPTARG}.raw"
+          ;;
+        : )
+          echo "Invalid Option: -${OPTARG} requires an argument" 1>&2
+          exit 1
+          ;;
+      esac
+    done
+    shift $((OPTIND-1))
```
Comment (andrewrynhard) [inline hack/installer/entrypoint.sh]: This is a workaround until #1389 is merged. Just allows the new Makefile layout to output the same artifact names as before.

File: Makefile [hunk 5]
```diff
@@ -41,250 +25,224 @@ COMMON_ARGS += --build-arg=TOOLS=$(TOOLS)
...
+## Creating a Builder Instance
 
-all: builddeps rootfs initramfs kernel osctl-linux osctl-darwin installer container
+The build process makes use of features not currently supported by the default
+builder instance (docker driver). To create a compatible builder instance, run:
 
-.PHONY: builddeps
-builddeps: gitmeta
+```
+docker buildx create --driver docker-container --name local --use
```
Comment (smira) [inline Makefile]: should add ` --buildkitd-flags '--allow-insecure-entitlement security.insecure'` here to allow unit-tests

Comment (andrewrynhard) [reply Makefile]: Thanks. 👍

**Pull request merged**